### ISSUE TITLE
Safari parseBlob failure

### ISSUE THREAD

**Issue by bradleat** [Borewit/music-metadata#2221]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### music-metadata version

10.2.0

### Current Behavior

When parsing a blob with Safari, I get the following error:

```
TypeError: ReadableStreamBYOBReader needs a ReadableByteStreamController
```

### Attached audio sample?

- [X] I have provided sufficient information to reproduce the issue

**Comment by bradleat** [comment]
I've tried using:

https://github.com/MattiasBuelens/web-streams-polyfill

It doesn't work.

**Comment by Borewit** [comment]
Do you have the same issue [here](https://audio-tag-analyzer.netlify.app/) using Safari?

A workaround is:

```js

import {parseBuffer} from 'music-metadata';

let blob; // Assume this Blob is assigned

const metadata = await parseBuffer(await blob.bytes()), {mimeType: blob.type});
```

Ref: 
- [blob.bytes()](https://developer.mozilla.org/en-US/docs/Web/API/Blob/bytes)
- [mm.parseBuffer()](https://github.com/Borewit/music-metadata?tab=readme-ov-file#parsebuffer-function)

**Comment by Borewit** [comment]
Is there is any news, feel free to re-open

**Comment by jonluca** [comment]
![Image](https://github.com/user-attachments/assets/cba06a5a-b90b-442f-889b-68510759e320)

Yes, the issue occurs there as well. The proposed workaround works though, for what its worth.

**Comment by Borewit** [comment]
I will try to switch from BYOB (bring your own buffer)/byte stream to the default stream.

Ref: https://developer.mozilla.org/docs/Web/API/ReadableStream/ReadableStream

**Comment by Borewit** [comment]
@jonluca can please retry here: https://fallback-on-default-reader--audio-tag-analyzer.netlify.app/

**Comment by jonluca** [comment]
![Image](https://github.com/user-attachments/assets/d5a922e5-96c4-4ed6-8cc8-d29a4890f6e2)

Same issue

**Comment by Borewit** [comment]
Do you mind to try one more time @jonluca ? GUI should show v10.8.0-beta.**3**", otherwise please reload your browser.

**Comment by jonluca** [comment]
![Image](https://github.com/user-attachments/assets/d8a70fc8-c6fa-4cc4-9d00-e818680ccd7b)

New error this time haha

**Comment by jonluca** [comment]
![Image](https://github.com/user-attachments/assets/2ab67649-a240-4324-8250-188afc85e370)

Same issue 

A way to debug might be to use bun - 

```js
import { parseBlob } from "music-metadata";
import { promises as fs } from "fs";

const data = await fs.readFile("~/Downloads/bohemian-rhapsody.mp3");
const blob = new Blob(data);
const metadata = await parseBlob(blob);
console.log(metadata);
```

```bash
bun test.mjs
```

Returns this exception

![Image](https://github.com/user-attachments/assets/3e80d9db-b6ed-44c7-bbb0-d44efac197bf)

**Comment by Borewit** [comment]
That was a pretty dumb attempt, as this is the the same `TypeError` error we had in the beginning.
This `TypeError` should be caught to fallback on [StreamDefaultReader](https://developer.mozilla.org/docs/Web/API/ReadableStreamDefaultReader).

You mind to try another time @jonluca, it may very very well now throw the Range out of bounds error. If you can please indicate where that occurs.

**Comment by jonluca** [comment]
I dont believe the site has an updated build - it's the same error as earlier.

However testing with main with bun seems to be working? 

I've created a PR that adds a test suite for bun, since it uses JavaScriptCore under the hood, the same engine as Safari. There seems to be a timeout with webstreams in JSC in WavPack but otherwise seems fine. 

https://github.com/Borewit/music-metadata/pull/2348

**Comment by Borewit** [comment]
> I dont believe the site has an updated build - it's the same error as earlier.

I was convinced I merged the [music-metadata v10.8.2](https://github.com/Borewit/music-metadata/releases/tag/v10.8.2), but apparently not. Now it is in via https://github.com/Borewit/audio-tag-analyzer/pull/1116. 

Relevant change is: https://github.com/Borewit/peek-readable/pull/794/commits/21b34df5cc3fa14360d08158f5eb6871a9bd4e2b

> I've created a PR that adds a test suite for bun, since it uses JavaScriptCore under the hood, the same engine as Safari. There seems to be a timeout with webstreams in JSC in WavPack but otherwise seems fine.

Awesome!

**Comment by sleepyfran** [comment]
Not sure if you folks are keeping the issue open for a specific reason, but tried the latest version of the library and Audio Tag Analyzer now and it works wonderfully!

<img width="400" alt="Image" src="https://github.com/user-attachments/assets/2e7aab60-1db0-4edc-8c73-d063e6fccceb" />
